.gitlab-ci.yml

```diff
@@ -44,7 +44,7 @@ builder-image:
     SDK_URL: https://bitcoincore.org/depends-sources/sdks
     XCODE_VERSION: "12.2"
     XCODE_BUILD_ID: 12B45b
-    MAKEJOBS: -j4
+    MAKEJOBS: -j8
```

Comment:
Would it be possible to make this dynamic based on the system itself?

Reply:
There is no way to assign a result of a bash script to a variable afaik. But we could do
```diff
diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 6283a1a45f..e1d2143956 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -44,9 +44,10 @@ builder-image:
     SDK_URL: https://bitcoincore.org/depends-sources/sdks
     XCODE_VERSION: "12.2"
     XCODE_BUILD_ID: 12B45b
-    MAKEJOBS: -j8
   before_script:
+    - export MAKEJOBS="-j$(nproc)"
     - echo HOST=$HOST
+    - echo MAKEJOBS=$MAKEJOBS
     - |
       if [ "$HOST" = "x86_64-apple-darwin" ]; then
         mkdir -p depends/SDKs
```
or simply
```diff
diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 6283a1a45f..bd5c877dfa 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -44,7 +44,6 @@ builder-image:
     SDK_URL: https://bitcoincore.org/depends-sources/sdks
     XCODE_VERSION: "12.2"
     XCODE_BUILD_ID: 12B45b
-    MAKEJOBS: -j8
   before_script:
     - echo HOST=$HOST
     - |
@@ -63,7 +62,7 @@ builder-image:
         fi
       fi
   script:
-    - make $MAKEJOBS -C depends HOST=$HOST $DEP_OPTS
+    - make -j$(nproc) -C depends HOST=$HOST $DEP_OPTS
   cache:
     # Let all branches share the same cache, which is ok because the depends subsystem is able to handle this properly (it works with hashes of all scripts)
     key:
```

Reply:
2nd options is no go bcs there are make deps, make, make check,  and make check-symbola

Reply:
> 2nd options is no go bcs there are make deps, make, make check, and make check-symbola

Not really. This template is used for jobs at depends stage only and its variables (including env) have a scope of a single job. Other stages use `$MAKEJOBS` set in `00_setup_env.sh` (which is included via `matrix.sh` in `.base-template`).